code/game/objects/items/devices/floor_painter.dm

```diff
@@ -0,0 +1,77 @@
+// Floor painter
+
+/obj/item/device/floor_painter
+	name = "floor painter"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "labeler1"
+	item_state = "flight"
+	
+	var/floor_icon
+	var/floor_state = "floor"
+	var/floor_dir = SOUTH
+	
+	var/list/allowed_states = list("arrival", "arrivalcorner", "bar", "barber", "blackcorner", "blue", "bluecorner", 
+		"bluefull", "bluered", "blueyellow", "blueyellowfull", "bot", "brown", "browncorner", "browncornerold", "brownold",
+		"cafeteria", "caution", "cautioncorner", "chapel", "cmo", "dark", "delivery", "escape", "escapecorner", "floor",
+		"freezerfloor", "gcircuit", "green", "greenblue", "greenbluefull", "greencorner", "greenfull", "greenyellow",
+		"greenyellowfull", "grimy", "loadingarea", "neutral", "neutralcorner", "neutralfull", "orange", "orangecorner",
+		"orangefull", "plaque", "purple", "purplecorner", "purplefull", "rampbottom", "ramptop", "red", "redblue", "redbluefull",
+		"redcorner", "redfull", "redgreen", "redgreenfull", "redyellow", "redyellowfull", "warning", "warningcorner", "warnwhite", 
+		"warnwhitecorner", "white", "whiteblue", "whitebluecorner", "whitebluefull", "whitebot", "whitecorner", "whitedelivery", 
+		"whitegreen", "whitegreencorner", "whitegreenfull", "whitehall", "whitepurple", "whitepurplecorner", "whitepurplefull", 
+		"whitered", "whiteredcorner", "whiteredfull", "whiteyellow", "whiteyellowcorner", "whiteyellowfull", "yellow", 
+		"yellowcorner", "yellowcornersiding", "yellowsiding")
+
+/obj/item/device/floor_painter/afterattack(var/atom/A, var/mob/user, proximity, params)
+	if(!proximity)
+		return
+	
+	var/turf/simulated/floor/plasteel/F = A
+	if(!istype(F))
+		to_chat(user, "<span class='warning'>\The [src] can only be used on station flooring.</span>")
+		return
+	
+	F.icon_state = floor_state
+	F.icon_regular_floor = floor_state
+	F.dir = floor_dir
+
+/obj/item/device/floor_painter/attack_self(var/mob/user)
+	if(!user)
+		return 0
+	user.set_machine(src)
+	interact(user)
+	return 1
+
+/obj/item/device/floor_painter/interact(mob/user as mob)
+	if(!floor_icon)
+		floor_icon = icon('icons/turf/floors.dmi', floor_state, floor_dir)
+	user << browse_rsc(floor_icon, "floor.png")
```

Comment:
The chem-master interface takes forever to pop up because of just a few simple *pills*--this is a lot more icons than that, so this is likely to cause even *more* lag.